src/reducers/explorer.reducer.ts

```diff
@@ -232,23 +232,16 @@ export default function explorerReducer(state = initialState, action: AnyAction)
             let itemNameValidation: IItemNameValidation = initialState.itemNameValidation;
             let tree: Nullable<IProjectItem> = state.tree;
 
-            // create array to be inserted into state
-            const objectArray: IProjectItem[] = importPathArray
-                // add '/' to each string
-                .map((path: string) => path.startsWith('/') ? path : '/' + path)
-                .map((path: any) => path.split('/').slice(1))
-                .reduce((children: any, path: any, idx: number) => insert(children, path, importSourceArray[idx]), <IProjectItem[]>[]);
-
             // add new item to the tree
             const [newTree, replacedTargetItem] = updateItemInTree(
                 state.tree,
                 parentId,
-                i => ({ ...i, children: sortProjectItems(i.children.concat(objectArray)) })
+                i => ({ ...i, children: sortProjectItems(i.children.concat(items)) })
```

Comment:
There is one bottleneck here - what happens if item with same name already exist? 
I'd say where we replace existing items or show a warning that it should be deleted.

Reply:
items with the same name are replaced now